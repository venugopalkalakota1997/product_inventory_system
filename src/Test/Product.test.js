import React from 'react';
import { MemoryRouter as Router, withRouter } from 'react-router-dom'
import {mount} from 'enzyme';
import {shallow} from 'enzyme';
import Product from '../compoents/Product';
import {configure} from 'enzyme'
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter:new Adapter()});

it('Render',()=>{
  const wrapper=shallow(<Product/>);
  const h=wrapper.find("h2");
  const result=h.text();

  expect(result).toBe("");
})

test('Product', () => {
    expect(true).toBeTruthy();
  });


it('Button ',()=>{
  const wrapper=mount(<button type="submit">Login</button>);
  const input=wrapper.find('button');
  const result=input.text();
  expect(result).toBe('Login')
  expect(input.prop('type')).toEqual('submit');
})
test('mathes Snapshot',()=>{
  const tree=renderer.create(<Router><Product/></Router>).toJSON()
  expect(tree).toMatchSnapshot();
})