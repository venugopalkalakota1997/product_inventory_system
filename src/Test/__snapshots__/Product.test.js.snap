// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes Snapshot 1`] = `
<div
  className="individual"
>
  <img
    alt="imagenotfound"
    className="image"
  />
  <h2 />
  <p />
  <p>
    Category :
     
  </p>
  <p
    className="description"
  >
    Description :
     
  </p>
  <p>
    <span
      style={
        Object {
          "float": "left",
          "padding": "0px 2px",
        }
      }
    >
      Price:Rs.
        
    </span>
    <span
      style={
        Object {
          "float": "right",
          "padding": "0px 32px",
        }
      }
    >
      Quantity:
    </span>
  </p>
  <span>
    <br />
    <br />
  </span>
</div>
`;
